koji-offsite-backup/osg-koji-offsite-backup

```diff
@@ -119,205 +141,137 @@ def csbacktick(*args, **kwargs):
     return outerr
 
 
-def remote_run_command(commandstr):
-    return sbacktick(['ssh', '-i', KEY, REMOTE_LOGIN_HOST, commandstr])
-
-
-def remote_run_script(script):
-    if isinstance(script, list) or isinstance(script, tuple):
-        script = "\n".join(script)
-
-    if DRYRUN:
-        print "---\n%s\n---" % script
-        return 0, ''
-
-    proc = Popen(['ssh', '-i', KEY, REMOTE_LOGIN_HOST, '/bin/bash -s'], stdin=PIPE, stdout=PIPE, stderr=STDOUT)
-    outerr = proc.communicate(script)[0]
-
-    return proc.returncode, outerr
+def rsync_with_retry(src_path, remote_path, delete=None):
+    rsync_opts = [
+        "--recursive",
+        "--stats",
+        "--rsh=ssh -i %s" % KEY,
+    ]
 
-
-def rsync_with_retry(src_path, remote_path, link=None):
-    rsync_opts = ['--recursive', '--compress', '--quiet', '--rsh=ssh -i %(KEY)s' % globals()]
-
-    if link:
-        rsync_opts.append('--link-dest=%s' % os.path.relpath(link, remote_path))
+    if delete:
+        rsync_opts.append("--%s" % delete)
 
     dest_path = "%s:%s" % (REMOTE_LOGIN_HOST, remote_path)
     count = 0
+    text = ""
     while True:
         try:
-            csbacktick(['rsync'] + rsync_opts + [src_path + '/', dest_path])
-            return
-        except CommandError, e:
+            if count > 0:
+                text += "Retry %d" % count
+            text += "-" * 60
```

Comment:
are there newlines between these `-----` bars?

Reply:
No but there oughta be.

I never noticed because it never triggered :-)

Reply:
I don't need to add a newline after the `"-" * 60` because the rsync output already starts with a newline (dunno why, but it does). I added a few newlines here and there though.